src/include/rbd/librbd.h

```diff
@@ -118,11 +118,11 @@ typedef struct {
 #define RBD_MAX_IMAGE_NAME_SIZE 96
 #define RBD_MAX_BLOCK_NAME_SIZE 24
 
-#define RBD_SNAP_CREATE_SKIP_QUIESCE		1 << 0
-#define RBD_SNAP_CREATE_IGNORE_QUIESCE_ERROR	1 << 1
+#define RBD_SNAP_CREATE_SKIP_QUIESCE		1u << 0
```

Comment:
While I don't think this will cause any issues, what was the warning here? I ask just because a few lines up there is "RBD_FLAG_OBJECT_MAP_INVALID (1 << 0)" and that wasn't changed.

Reply:
"warning: shifting a negative signed value is undefined" Though it was mentioned as being the result of a macro expansion, so I think it didn't flag the others since those macros didn't get referenced.

(I'll just make the analogous change there for consistency.)

Reply:
Odd -- it's not a negative and all references to that Clang warning that I googled showed actual negative constants being shifted.

Reply:
(Alternatively maybe it just doesn't complain about zero shifts.)

Reply:
I see what it is now. Lack of parentheses in the definition.

Reply:
All right. 829623f3f6a291224c9f1b3da53b2a627d96a2b4 should be the right fix for this.